corporate/views/remote_billing_page.py

```diff
@@ -0,0 +1,103 @@
+import logging
+from typing import Dict, Union
+
+from django.conf import settings
+from django.core import signing
+from django.http import HttpRequest, HttpResponse, HttpResponseRedirect
+from django.urls import reverse
+from django.utils.translation import gettext as _
+from django.views.decorators.csrf import csrf_exempt
+
+from zerver.decorator import zulip_login_required
+from zerver.lib.exceptions import JsonableError, OrganizationOwnerRequiredError
+from zerver.lib.push_notifications import uses_notification_bouncer
+from zerver.lib.remote_server import send_to_push_bouncer
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.validator import check_int
+from zerver.models import UserProfile
+from zilencer.models import RemoteZulipServer
+
+billing_logger = logging.getLogger("corporate.stripe")
+
+
+SIGNED_ACCESS_TOKEN_VALIDITY_IN_SECONDS = 60 * 60 * 24
+
+
+@csrf_exempt
+@has_request_variables
+def remote_server_billing_entry(
+    request: HttpRequest,
+    remote_server: RemoteZulipServer,
+    user_uuid: str = REQ(),
+    realm_id: int = REQ(json_validator=check_int),
+) -> HttpResponse:
+    identity_dict = {
+        "realm_id": realm_id,
+        "user_uuid": user_uuid,
+        "remote_server_uuid": str(remote_server.uuid),
+    }
+    signed_identity_dict = signing.dumps(identity_dict)
+
+    billing_access_url = settings.ROOT_DOMAIN_URI + reverse(
+        remote_server_billing_ui, args=[signed_identity_dict]
+    )
+    return json_success(request, data={"billing_access_url": billing_access_url})
+
+
+def remote_server_billing_ui(
+    request: HttpRequest, signed_billing_access_token: str
+) -> HttpResponse:
+    try:
+        identity_dict = signing.loads(
+            signed_billing_access_token, max_age=SIGNED_ACCESS_TOKEN_VALIDITY_IN_SECONDS
+        )
+    except signing.SignatureExpired:
+        raise JsonableError(_("Billing access token expired."))
+    except signing.BadSignature:
+        raise JsonableError(_("Invalid billing access token."))
+
+    realm_id = identity_dict["realm_id"]
+    user_uuid = identity_dict["user_uuid"]
+    remote_server_uuid = identity_dict["remote_server_uuid"]
+
+    try:
+        remote_server = RemoteZulipServer.objects.get(uuid=remote_server_uuid)
+    except RemoteZulipServer.DoesNotExist:
+        raise JsonableError(_("Invalid remote server."))
+
+    remote_server_info = {
+        "remote_server_uuid": remote_server_uuid,
+        "remote_server_hostname": remote_server.hostname,
+        "remote_server_contact_email": remote_server.contact_email,
+        "remote_server_plan_type": remote_server.plan_type,
+    }
+    return json_success(
+        request,
+        data={
+            "user_uuid": user_uuid,
+            "realm_id": realm_id,
+            "remote_server_info": remote_server_info,
+        },
+    )
```

Comment:
Am I correct that what we're doing here is just returning a JSON blob to confirm things are working, but this would in the future just be rendering a template with these fields as context?

Reply:
Correct. Now I replaced this response with a dummy html template that displays some user and server info